src/ng/compile.js

```diff
@@ -174,17 +174,22 @@
  *   `localName` property on the widget scope. The `name` is read from the parent scope (not
  *   component scope).
  *
- * * `=` or `=attr` - set up bi-directional binding between a local scope property and the
- *   parent scope property of name defined via the value of the `attr` attribute. If no `attr`
- *   name is specified then the attribute name is assumed to be the same as the local name.
- *   Given `<widget my-attr="parentModel">` and widget definition of
- *   `scope: { localModel:'=myAttr' }`, then widget scope property `localModel` will reflect the
- *   value of `parentModel` on the parent scope. Any changes to `parentModel` will be reflected
- *   in `localModel` and any changes in `localModel` will reflect in `parentModel`. If the parent
- *   scope property doesn't exist, it will throw a NON_ASSIGNABLE_MODEL_EXPRESSION exception. You
- *   can avoid this behavior using `=?` or `=?attr` in order to flag the property as optional. If
- *   you want to shallow watch for changes (i.e. $watchCollection instead of $watch) you can use
- *   `=*` or `=*attr` (`=*?` or `=*?attr` if the property is optional).
+ * * `=` or `=attr` - set up a bidirectional binding between a local scope property and an expression
+ *   passed via the attribute `attr`. The expression is evaluated in the context of the parent scope.
+ *   If no `attr` name is specified then the attribute name is assumed to be the same as the local
+ *   name. Given `<widget my-attr="parentModel">` and the widget definition `scope: { localModel:
+ *   '=myAttr' }`, the property `localModel` on the widget's scope will reflect the value of
+ *   `parentModel` on the parent scope. Changes to `parentModel` will be reflected in `localModel`
+ *   and vice versa. Optional attributes should be marked as such with a question mark: `=?` or
+ *   `=?attr`. If the binding expression is non-assignable, or if the attribute isn't optional and
+ *   doesn't exist, an exception ({@link error/$compile/nonassign `$compile:nonassign`}) will be
+ *   thrown on discovering changes to the local value since it'll be impossible to sync it with the
+ *   parent scope in this case. By default, the {@link ng.$rootScope.Scope#$watch `$watch`} method is
+ *   used for tracking changes, and the equality check is done by reference. However, if an object
```

Comment:
Personally, I find it clearer to say that the equality check is based on object identity. I guess it means the same though.